docs/sdk/src/reference_docs/cli.rs

```diff
@@ -1,7 +1,104 @@
-//! # Command Line Arguments
+//! # Substrate CLI
 //!
+//! Let's see some examples of typical CLI arguments used when setting up and running a
+//! Substrate-based blockchain. We use the [`substrate-node-template`](https://github.com/substrate-developer-hub/substrate-node-template)
+//! on these examples.
 //!
-//! Notes:
+//! #### Checking the available CLI arguments
+//! ```bash
+//! ./target/debug/node-template --help
+//! ```
+//! - `--help`: Displays the available CLI arguments.
 //!
-//! - Command line arguments of a typical substrate based chain, and how to find and learn them.
-//! - How to extend them with your custom stuff.
+//! #### Starting a Local Substrate Node in Development Mode
+//! ```bash
+//! ./target/release/node-template \
+//! --dev
+//! ```
+//! - `--dev`: Runs the node in development mode, using a pre-defined development chain
```

Comment:
Should link to [this](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/client/cli/src/params/shared_params.rs#L40) Rust type to avoid any repetition or inaccurate info. 

In other words, there should only be one source of truth for what `--dev` means, which is [here](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/client/cli/src/params/shared_params.rs#L40). A short paraphrasing is okay, but let's link to it. 

Same can be applied for technically every other arg mentioned here. Please make sure at least the important ones are linked to the proper Rust type, and that the Rust type is well documented.